lib/ansible/modules/cloud/kubevirt/kubevirt_vm_status.py

```diff
@@ -132,127 +132,94 @@
+except ImportError as exc:
+    class KubernetesException(Exception):
+        pass
+
 
-from kubernetes.client.rest import ApiException
-from kubernetes.config import kube_config, ConfigException
+VM_ARG_SPEC = {
+    'state': {'type': 'str', 'required': True, 'choices': ['running', 'stopped']},
+}
 
 
-class KubeVirtVMStatus(K8sVirtAnsibleModule):
+class KubeVirtVM(KubernetesRawModule):
     def __init__(self, *args, **kwargs):
-        super(KubeVirtVMStatus, self).__init__(*args, **kwargs)
-        self._api_client = None
+        super(KubeVirtVM, self).__init__(*args, **kwargs)
 
     @property
     def argspec(self):
         """ argspec property builder """
-        argspec = copy.deepcopy(NAME_ARG_SPEC)
-        argspec.update(copy.deepcopy(AUTH_ARG_SPEC))
-        argspec['name']['required'] = True
-        argspec['namespace']['required'] = True
-        argspec['api_version']['default'] = 'v1alpha2'
-        argspec['state'] = dict({
-            'required': False, 'type': 'str',
-            'choices': list(['running', 'stopped']), 'default': 'running'
-        })
-        return argspec
+        argument_spec = copy.deepcopy(COMMON_ARG_SPEC)
+        argument_spec.update(copy.deepcopy(AUTH_ARG_SPEC))
+        argument_spec.update(VM_ARG_SPEC)
+        return argument_spec
 
     def execute_module(self):
         """ Module execution """
-        api_client = self.__authenticate()
-        api_instance = kubernetes.client.CustomObjectsApi(
-            api_client=api_client)
-        group = 'kubevirt.io'
-        plural = 'virtualmachines'
-        version = self.params.get('api_version')
-        namespace = self.params.get('namespace')
-        name = self.params.get('name')
-        state = True if self.params.get('state') == 'running' else False
-        body = dict()
-        body['spec'] = dict({'running': state})
-
-        try:
-            exists = api_instance.get_namespaced_custom_object(
-                group, version, namespace, plural, name)
-            current_state = exists.get('spec').get('running')
-
-            if current_state == state:
-                self.exit_json(changed=False)
+        definition = self.resource_definitions[0]
+        self.client = self.get_api_client()
 
-            api_response = api_instance.patch_namespaced_custom_object(
-                group, version, namespace, plural, name, body)
-            self.exit_json(changed=True, result=api_response)
-        except ApiException as exc:
-            self.fail_json(msg='Failed to manage requested object',
-                           error=exc.reason)
+        state = self.params.get('state')
+        resource_version = self.params.get('resource_version')
+        name = definition['metadata']['name']
+        namespace = definition['metadata'].get('namespace')
+        api_version = self.params.get('apiVersion', 'kubevirt.io/v1alpha2')
+        kind = self.params.get('kind', 'VirtualMachine')
```

Comment:
Why do we need this. We know what is the kind for this piece of code.

Reply:
Right, I guess we don't have to provide user the option to override it.